### PR TITLE
#101 Simplify configuration

### PR DESCRIPTION
**Pull request by epeee**
a very first version, I will try to add some tests soon.

### CODE REVIEW

File: src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        private final ProcessRunner processRunner;
+        private volatile String branchName;
+        private static final Object SYNC = new Object();
+
+        public GitStatus(Project project) {
+             processRunner = Exec.getProcessRunner(project.getRootDir());
+        }
+
+        public void setBranchName(String branchName) {
```
Comment (szczepiq) [inline src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java]: Let's make it even simpler :)

1. I suggest we get rid of the 'setBranchName' method from here
2. We can create a new task type 'GitCheckOutTask' task that has 'setRev' method which is the target revision we want to check out.
3. New task type 'GitCheckOutTask' will be used for creating 'checkOutBranch' task  in GitPlugin
4. TravisPlugin already applies GitPlugin so travis plugin can configure checkOutBranch.setRev with the value acquired from env variable

This way, we don't offer way to set branch and things are simpler.

File: src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java [hunk 2]
```diff
@@ -35,7 +37,7 @@ public void apply(Project project) {
         boolean isPullRequest = pr != null && !pr.trim().isEmpty() && !pr.equals("false");
 
         conf.getBuild().setCommitMessage(System.getenv("TRAVIS_COMMIT_MESSAGE"));
-        conf.getBuild().setBranch(System.getenv("TRAVIS_BRANCH"));
+        gitStatus.setBranchName(System.getenv("TRAVIS_BRANCH"));
```
Comment (szczepiq) [inline src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java]: This is a little bit risky. If some other plugin requests the branch during Gradle's configuration time, and before travis plugin is applied then some objects might get wrong value of the branch. I suggest to apply my earlier suggestion and get rid of setBranchName method.

File: src/main/groovy/org/mockito/release/internal/gradle/ContinuousDeliveryPlugin.java [hunk 3]
```diff
@@ -206,7 +206,8 @@ public void execute(final ReleaseNeededTask t) {
 
                 lazyConfiguration(t, new Runnable() {
                     public void run() {
-                        t.setBranch(conf.getBuild().getBranch());
+                        String branch = project.getExtensions().getByType(GitStatusPlugin.GitStatus.class).getBranch();
```
Comment (szczepiq) [inline src/main/groovy/org/mockito/release/internal/gradle/ContinuousDeliveryPlugin.java]: Nice!

File: src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package org.mockito.release.internal.gradle;
+
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.mockito.release.exec.Exec;
+import org.mockito.release.exec.ProcessRunner;
+
+public class GitStatusPlugin implements Plugin<Project> {
```
Comment (szczepiq) [inline src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java]: Great new plugin! Javadocs please :)

File: src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java (hunk 2 above)
Comment (epeee) [inline src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java]: k, thx for pointing that out. I will apply your suggestion later on today.

**Comment by epeee** [comment]
@szczepiq ad 3, joining the team, yes, would be great and would make collaboration easier, thx.

File: src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java (hunk 2 above)
Comment (szczepiq) [inline src/main/groovy/org/mockito/release/internal/gradle/TravisPlugin.java]: Awesome!

**Comment by szczepiq** [comment]
>@szczepiq ad 3, joining the team, yes, would be great and would make collaboration easier, thx.

Done. Welcome!

**Comment by szczepiq** [comment]
@epeee, please create a dev branch in release-tools-project that you will work on. It will make it easier for us to review and make changes directly on your branch.

File: src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java (hunk 1 above)
Comment (epeee) [inline src/main/groovy/org/mockito/release/internal/gradle/GitStatusPlugin.java]: Thx, your suggestion makes it way cleaner. :+1:

**Comment by epeee** [comment]
@szczepiq I will create a new PR using my dev branch from release-tools-project.

**Comment by mstachniuk** [comment]
Closed because of new PR: https://github.com/mockito/mockito-release-tools/pull/109

**Pull request closed without merging**